### PR TITLE
[CA-1318] Sr add integration test to check roles on created google project

### PR DESCRIPTION
**Pull request by s-rubenstein**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/project/Dependencies.scala
Comment (s-rubenstein) [inline automation/project/Dependencies.scala]: Should I do a full publish here? I assume so, but wanted to check with folks first? (Or should it be publish a snap version based on new v2_workspaces workbench libs branch?)

Comment (marctalbott) [reply automation/project/Dependencies.scala]: The v2_workspaces `workbench-libs` feature branch just updates the `rawls-model` version used in service test. You should use the most recent version of the `google` library that was published after you merged your `workbench-libs` PR.

File: automation/project/Dependencies.scala
Comment (marctalbott) [inline automation/project/Dependencies.scala]: Is this version update necessary? Your workbench-libs PR only updated the `google` library.

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala
Comment (marctalbott) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: nit: believe this comment is misleading. Terra billing project name requirements are less strict now: https://github.com/broadinstitute/rawls/blob/48bee8ad0bc2eefc4dcf254dc3b282133d3b9006/model/src/main/scala/org/broadinstitute/dsde/rawls/StringValidationUtils.scala#L44

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala
Comment (marctalbott) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: Some of these policy bindings are constant across test runs right? Like the permissions that the Rawls SA has? Might be worth making the checks more specific for those roles.

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala
Comment (marctalbott) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: We could add a `withCleanup` block to ensure that the workspace and billing project are cleaned up here in case the test fails

See example in Sam automation tests here `https://github.com/broadinstitute/sam/blob/fa74f750b4e7652b28fa4287aaff0fbd9c7c0450/automation/src/test/scala/org/broadinstitute/dsde/test/api/SamApiSpec.scala#L231`

**Review by marctalbott — APPROVED** [review]
Left some notes on what to do with the workbench-libs dependencies.

Everything else is up to you -- they're suggested improvements to be made in an ideal world, but they're nothing I would block the PR over. Given the relatively lower priority of this ticket compared to other PPW work, it probably makes sense to skip any of the suggested changes if they take more than a couple minutes to make

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala
Comment (s-rubenstein) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: Weirdly didn't see the Rawls SA on there. I think its permissions are at the folder level.

**Comment by s-rubenstein** [comment]
Jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala
Comment (andy7i) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: there's a small chance that someone tries to run this test locally (against a non-fiab env, like dev) in which case this would fail. The Right thing to do would be to get this from a config, but we could also just add a comment explaining that this test is only meant to run against a fiab or check if the member ends with `@terra-kernel-k8s.iam.gserviceaccount.com` or something like that.

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala
Comment (andy7i) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:113]: do we know that these 12 are the correct permissions, or are we going with the assumption that they are correct and we want to persist the permissions that we have now?

**Review by andy7i — APPROVED** [review]
awesome test!
Added a couple questions, but not blocking 👍

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala
Comment (s-rubenstein) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:113]: We are looking to just persist the permissions we have right now.

Comment (andy7i) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:113]: SGTM! just wanted to make that explicit

**Comment by s-rubenstein** [comment]
Jenkins retest

**Comment by s-rubenstein** [comment]
Jenkins retest

**Comment by s-rubenstein** [comment]
Jenkins retest

**Pull request merged**
